### ISSUE TITLE
The variable `foo` is not used anywhere in the query! Check for typos.

### ISSUE THREAD

**Issue by xRodzMC** [poggit/libasynql#86]
### Stack Error Message
```
Error: The variable kills is not used anywhere in the query! Check for typos.
File: virions/src/poggit/libasynql/generic/GenericStatementImpl
Line: 187
```
### Table
```sql
-- #{ setKills
-- #	:player_name string
-- #    :kills int
INSERT OR IGNORE INTO Practice (player_name, kills) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET kills = :kills WHERE player_name = :player_name;
-- #}
-- #{ getKills
-- #	:player_name string
SELECT kills FROM Practice WHERE player_name = :player_name; 
-- #}
```

### Code
```php
<?php

declare(strict_types=1);

namespace PrideCore\Utils;

use poggit\libasynql\libasynql;
use poggit\libasynql\DataConnector;
use pocketmine\utils\SingletonTrait;
use pocketmine\utils\Config;
use PrideCore\Core;

class Database {
    use SingletonTrait;
    
    public function getDatabase() : DataConnector{
        return libasynql::create(Core::getInstance(), $this->getConfig()->get("database"), [
			"sqlite" => "sqlite.sql",
			"mysql" => "mysql.sql",
		]);
    }
    
    public function getConfig() : Config{
        return new Config(Core::getInstance()->getDataFolder() . "database.yml", Config::YAML);
    }
    
    public function init() : void{
        $this->getDatabase()->executeGeneric("init");
        $this->getDatabase()->waitAll();
    }
}
```
# Screenshots or Links
- [Full `sqlite.sql` Statement](https://pastebin.com/HcxTeaME)
### Discord Support Images
People who are helping me to indentify and fix the bug:
![Screenshot_2022_0912_221124](https://user-images.githubusercontent.com/100104970/189675972-dab657d5-b376-41c2-9c8e-41ce72a97785.jpg)

![Screenshot_2022_0912_220952](https://user-images.githubusercontent.com/100104970/189675638-97cb2419-f500-4895-ba90-5d596560d952.jpg)

**Comment by Endermanbugzjfc** [comment]
> 

`:kills` not used, as the error said.

**Comment by Endermanbugzjfc** [comment]
# TL;DR
A variable is treated as unused if you do not use it in the first statement of `-- #&`.

**Comment by xRodzMC** [comment]
Check full statements provided in the attachements.

**Comment by SOF3** [comment]
cannot be reproduced with the latest master by taking your SQL code above and prepending `-- #!sqlite` in front. Can you upload your file directly?

**Comment by xRodzMC** [comment]
> cannot be reproduced with the latest master by taking your SQL code above and prepending `-- #!sqlite` in front. Can you upload your file directly?

the full statement file `sqlite.sql` is uploaded on above.

Im using sqlite based to my config.

**Comment by Endermanbugzjfc** [comment]
> > cannot be reproduced with the latest master by taking your SQL code above and prepending `-- #!sqlite` in front. Can you upload your file directly?
> 
> 
> 
> the full statement file `sqlite.sql` is uploaded on above.
> 
> 
> 
> Im using sqlite based to my config.

You do not have `-- #!sqlite` at the heading. Your file is not even correct.

**Comment by SOF3** [comment]
please upload the exact file you had a problem with ,not copy paste. I suspect you have some invisible characters there.

**Comment by xRodzMC** [comment]
[Hello???](https://pastebin.com/HcxTeaME)

**Comment by xqwtxon** [comment]
# TL;DR
```sql
-- #!sqlite
-- #{ init
CREATE TABLE Rank (
	player_name TEXT PRIMARY KEY COLLATE NOCASE,
	rank INTEGER DEFAULT 0,
);
CREATE TABLE Practice (
    player_name TEXT PRIMARY KEY COLLATE NOCASE,
    kills INTEGER DEFAULT 0,
    deaths INTEGER DEFAULT 0,
    wins INTEGER DEFAULT 0,
    loses INTEGER DEFAULT 0,
    elo INTEGER DEFAULT 0
);
-- #&
CREATE INDEX Rank_rank ON Rank (rank);
CREATE INDEX Rank_banned ON Rank (banned);

CREATE INDEX Practice_kills ON Practice (kills);
CREATE INDEX Practice_deaths ON Practice (deaths);
CREATE INDEX Practice_wins ON Practice (wins);
CREATE INDEX Practice_loses ON Practice (loses);
CREATE INDEX Practice_elo ON Practice (elo);
-- #}
-- #{ setRank
-- #	:player_name string
-- #    :rank int
INSERT OR IGNORE INTO Rank (player_name, rank) VALUES (:player_name, 0);
-- #&
UPDATE Rank SET rank = :rank WHERE player_name = :player_name;
-- #}
-- #{ getRank
-- #	:player_name string
SELECT rank FROM Rank WHERE player_name = :player_name;
-- #}
-- #{ setKills
-- #	:player_name string
-- #    :kills int
INSERT OR IGNORE INTO Practice (player_name, kills) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET kills = :kills WHERE player_name = :player_name;
-- #}
-- #{ getKills
-- #	:player_name string
SELECT kills FROM Practice WHERE player_name = :player_name;
-- #}
-- #{ setDeaths
-- #	:player_name string
-- #    :deaths int
INSERT OR IGNORE INTO Practice (player_name, deaths) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET deaths = :deaths WHERE player_name = :player_name;
-- #}
-- #{ getDeaths
-- #	:player_name string
SELECT deaths FROM Practice WHERE player_name = :player_name;
-- #}
-- #{ setWins
-- #	:player_name string
-- #    :kills int
INSERT OR IGNORE INTO Practice (player_name, wins) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET wins = :wins WHERE player_name = :player_name;
-- #}
-- #{ getWins
-- #	:player_name string
SELECT wins FROM Practice WHERE player_name = :player_name;
-- #}
-- #{ setLoses
-- #	:player_name string
-- #    :kills int
INSERT OR IGNORE INTO Practice (player_name, loses) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET loses = :loses WHERE player_name = :player_name;
-- #}
-- #{ getLoses
-- #	:player_name string
SELECT loses FROM Practice WHERE player_name = :player_name;
-- #}
-- #{ setElo
-- #	:player_name string
-- #    :elo int
INSERT OR IGNORE INTO Practice (player_name, elo) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET elo = :elo WHERE player_name = :player_name;
-- #}
-- #{ getElo
-- #	:player_name string
SELECT elo FROM Practice WHERE player_name = :player_name;
-- #}
```

**Comment by SOF3** [comment]
@xqwtxon I said, do not copy-paste the file, including pastebin. Upload your file directly by dragging the file into this webpage, or by clicking the upload bar below the comment box.

**Comment by xRodzMC** [comment]
> @xqwtxon I said, do not copy-paste the file, including pastebin. Upload your file directly by dragging the file into this webpage, or by clicking the upload bar below the comment box.

![Screenshot_2022_0914_145645](https://user-images.githubusercontent.com/100104970/190082872-d79d7949-1f48-43a9-8498-b1afbedcd613.jpg)

**Comment by Leafres** [comment]
@xqwtxon Put it in a ZIP file

**Comment by xRodzMC** [comment]
> @xqwtxon Put it in a ZIP file

[mysql.sql.zip](https://github.com/poggit/libasynql/files/9565636/mysql.sql.zip)

**Comment by SOF3** [comment]
Your error is on these lines
```
-- #{ setWins
-- #	:player_name string
-- #    :kills int
INSERT OR IGNORE INTO Practice (player_name, wins) VALUES (:player_name, 0);
-- #&
UPDATE Practice SET wins = :wins WHERE player_name = :player_name;
-- #}
```
you wrote `:wins` instead of `:kills`

however I agree that the error message needs improvement to reflect which statement is wrong.
